AppCenter/AppCenter/Internals/Channel/MSOneCollectorChannelDelegate.m

```diff
@@ -209,16 +209,37 @@ - (BOOL)validateLogName:(NSString *)name {
 }
 
 - (BOOL)validateLogData:(MSCSData *)data {
-  NSDictionary<NSString *, NSString *> *properties = data.properties;
+  return [self validateProperties:data.properties];
+}
+
+- (BOOL)validateProperties:(NSDictionary<NSString *, id> *) properties {
+  BOOL valueIsAString;
   for (NSString *key in properties) {
-    if (![key isKindOfClass:[NSString class]] ||
-        ![properties[key] isKindOfClass:[NSString class]]) {
+    BOOL keyIsNSString = [key isKindOfClass:[NSString class]]; // is this actyally a case?!
```

Comment:
Do you actually want this comment here?  Maybe a more general comment on why this is or shouldn't be a case would be more helpful